app/viewmodels/SecurityStatementsViewModel.scala

```diff
@@ -16,9 +16,259 @@
+  }
+
+  private def generateRequestedStatementNotification(statementsForAllEoris: Seq[SecurityStatementsForEori])
+                                                    (implicit appConfig: AppConfig, messages: Messages
+                                                    ): HtmlFormat.Appendable = {
+    if (hasRequestedStatements(statementsForAllEoris)) {
+      new requestedStatements(linkComponent).apply(
+        url = appConfig.requestedStatements(SecurityStatement))
+    } else {
+      HtmlFormat.empty
+    }
+  }
+
+  private def generateCurrentStatements(statementsForAllEoris: Seq[SecurityStatementsForEori])
+                                       (implicit messages: Messages): HtmlFormat.Appendable = {
+    if (hasCurrentStatements(statementsForAllEoris)) {
+      HtmlFormat.fill(statementsForAllEoris.zipWithIndex.flatMap {
+        case (statementsForEori, historyIndex) if statementsForEori.currentStatements.nonEmpty =>
+          val eoriHeader = generateEoriHeader(historyIndex, statementsForEori)
+          val eomHeader = generateEomHeader(statementsForEori)
+
+          val pdfStatements = generateStatements(statementsForEori, historyIndex, isCsv = false)
+          val csvStatements = generateStatements(statementsForEori, historyIndex, isCsv = true)
+
+          Seq(eoriHeader, Some(pdfStatements), eomHeader, Some(csvStatements)).flatten
+
+        case _ => Seq.empty
+      })
+    } else {
+      generateNoStatementsParagraph
+    }
+  }
+
+  private def generateEoriHeader(historyIndex: Int, statementsForEori: SecurityStatementsForEori)
+                                (implicit messages: Messages): Option[HtmlFormat.Appendable] = {
+    if (historyIndex > 0) {
+      Some(h2Component(
+        msg = messages("cf.account.details.previous-eori", statementsForEori.eoriHistory.eori),
+        id = Some(s"historic-eori-$historyIndex"),
+        classes = "govuk-heading-s govuk-!-margin-bottom-1"))
+    } else {
+      None
+    }
+  }
+
+  private def generateEomHeader(statementsForEori: SecurityStatementsForEori)
+                               (implicit messages: Messages): Option[HtmlFormat.Appendable] = {
+    if (statementsForEori.currentStatements.exists(_.hasCsv)) {
+      Some(h2Component(
+        msg = messages("cf.security-statements.eom"),
+        classes = "govuk-heading-m govuk-!-padding-top-2"))
+    } else {
+      None
+    }
+  }
+
+  private def generateStatements(statementsForEori: SecurityStatementsForEori, historyIndex: Int, isCsv: Boolean)
+                                (implicit messages: Messages): HtmlFormat.Appendable = {
+    val statementContent = statementsForEori.currentStatements.zipWithIndex.flatMap {
+      case (statement, index) if isCsv != statement.hasPdf =>
+        Some(generateStatementRow(statement, historyIndex, index, isCsv))
+      case _ => None
+    }
+
+    dlComponent(
+      content = HtmlFormat.fill(statementContent),
+      classes = Some("govuk-summary-list statement-list"),
```

Comment:
@Neozxc  The Gov.UK Design system already has the [Summary list component ](https://design-system.service.gov.uk/components/summary-list/) already built you could import this instead which could reduce the need for custom components like `dtComponent` and `ddComponent`, making the code easier to manage.

Reply:
This has been addressed over chat.